src/services/signer_service.ts

```diff
@@ -173,12 +205,93 @@ export class SignerService {
+        while (true) {
+            logger.trace('watching for stuck transactions');
+            try {
+                // NOTE(oskar): we currently cancel all transactions that are in the
+                // "stuck" state. A better solution might be to unstick
+                // transactions one by one and observing if they unstick the
+                // subsequent transactions.
+                const stuckTransactions = await this._transactionEntityRepository.find({
+                    where: { status: TransactionStates.Stuck },
+                });
+                const targetGasPrice = await this._getGasPriceFromGasStationOrThrowAsync();
+                for (const tx of stuckTransactions) {
+                    try {
+                        await this._unstickTransactionAsync(
+                            tx,
+                            targetGasPrice.multipliedBy(new BigNumber(UNSTICKING_TRANSACTION_GAS_MULTIPLIER)),
+                        );
+                    } catch (err) {
+                        logger.error(`failed to unstick transaction ${tx.hash}`, { err });
+                    }
+                }
+            } catch (err) {
+                logger.error('failed to query stuck transactions', { err });
+            }
+            await utils.delay(SIGNER_POLLING_INTERVAL_IN_MS);
+        }
+    }
+    private async _unstickTransactionAsync(tx: TransactionEntity, gasPrice: BigNumber): Promise<string> {
+        const ethereumTxnParams: PartialTxParams = {
+            from: META_TXN_RELAY_ADDRESS,
+            to: META_TXN_RELAY_ADDRESS,
+            value: web3WrapperUtils.encodeAmountAsHexString(0),
+            nonce: tx.nonce,
+            chainId: CHAIN_ID,
+            gasPrice: web3WrapperUtils.encodeAmountAsHexString(gasPrice),
+            gas: web3WrapperUtils.encodeAmountAsHexString(ETH_TRANSFER_GAS_LIMIT),
+        };
+        const ethTx = new EthereumTx(ethereumTxnParams);
+        ethTx.sign(this._privateKeyBuffer, true);
+        const txHash = ethTx.hash() as Buffer;
+        const txHashHex = `0x${txHash.toString('hex')}`;
+        const transactionEntity = new TransactionEntity({
+            hash: txHashHex,
+            status: TransactionStates.Unsubmitted,
+            nonce: tx.nonce,
+            gasPrice: gasPrice.toString(),
+            metaTxnRelayerAddress: META_TXN_RELAY_ADDRESS,
+            expectedMinedInSec: EXPECTED_MINED_IN_S,
+        });
+        await this._transactionEntityRepository.save(transactionEntity);
+        const rawTx = `0x${ethTx.serialize().toString('hex')}`;
+        await this._web3Wrapper.sendRawPayloadAsync({
+            method: 'eth_sendRawTransaction',
+            params: [rawTx],
+        });
+        await this._updateTransactionEntityToSubmittedAsync(txHashHex);
+        transactionEntity.status = TransactionStates.Submitted;
+        await this._transactionEntityRepository.save(transactionEntity);
```

Comment:
Remove these two lines. We already update it's status with call to `_updateTransactionEntityToSubmittedAsync`.